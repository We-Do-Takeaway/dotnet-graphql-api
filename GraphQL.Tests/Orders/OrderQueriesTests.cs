using System;
using System.Threading.Tasks;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using WeDoTakeawayAPI.GraphQL.Model;
using Xunit;

namespace WeDoTakeawayAPI.GraphQL.Tests.Orders
{
    public class OrderQueriesTests: BaseTests
    {
        private async Task<Model.Order> CreateOrder()
        {
            ApplicationDbContext dbContext = GetDbContext();

            var orderId = Guid.Parse("56c1f9c7-54a5-4f3d-b967-1f4073415d08");

            var order = new Model.Order
            {
                Id = orderId,
                Name = "Fred",
                Address1 = "ad1",
                Address2 = "ad2",
                Town = "town",
                Postcode = "pcode",
                Phone = "1234",
                Email = "fred@d.d",
                DeliveryInstructions = "In the coal bunker",
                OwnerId = Guid.Parse("9940c08a-34b5-4e42-a736-bf9aec716f09"),
                CreatedAt = DateTime.Parse("08/18/2018 07:22:16")
            };

            var orderItem = new OrderItem
            {
                OrderId = orderId,
                ItemId = Guid.Parse("600DCA30-C6E2-4035-AD15-783C122D6EA4"),
                Quantity = 1,
                Name = "Plate of sausages",
                Description = "Big bowl of sausages"
            };

            await dbContext.Orders.AddAsync(order);
            await dbContext.OrderItems.AddAsync(orderItem);
            await dbContext.SaveChangesAsync();

            return order;
        }

        [Fact]
        public async Task Get_Order_For_Id()
        {
            Model.Order order = await CreateOrder();

            IExecutionResult result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                        query OrderById($id:ID!) {
	                        orderById(id: $id) {
                            id
                            name
                            address1
                            address2
                            town
                            postcode
                            phone
                            email
                            deliveryInstructions
                            items {
                              itemId
                              name
                              description
                              quantity
                            }
                          }
                        }
                    ")
                    .SetVariableValue(name: "id", value: order.Id.ToString())
                    .Create());

            result.MatchSnapshot();
        }

        [Fact]
        public async Task Get_Orders()
        {
            Model.Order order = await CreateOrder();

            IExecutionResult result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                        query Orders {
	                        orders {
                                id
                                name
                                address1
                                address2
                                town
                                postcode
                                phone
                                email
                                deliveryInstructions
                                items {
                                  itemId
                                  name
                                  description
                                  quantity
                                }
                            }
                        }
                    ")
                    .Create());

            result.MatchSnapshot();
        }
    }
}
