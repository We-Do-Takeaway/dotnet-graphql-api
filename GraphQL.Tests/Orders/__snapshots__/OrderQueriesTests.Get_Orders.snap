{
  "Label": null,
  "Path": null,
  "Data": null,
  "Errors": [
    {
      "Message": "The field `id` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 4,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "id",
        "responseName": "id",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    },
    {
      "Message": "The field `name` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 5,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "name",
        "responseName": "name",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    },
    {
      "Message": "The field `address1` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 6,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "address1",
        "responseName": "address1",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    },
    {
      "Message": "The field `address2` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 7,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "address2",
        "responseName": "address2",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    },
    {
      "Message": "The field `town` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 8,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "town",
        "responseName": "town",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    },
    {
      "Message": "The field `postcode` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 9,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "postcode",
        "responseName": "postcode",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    },
    {
      "Message": "The field `phone` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 10,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "phone",
        "responseName": "phone",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    },
    {
      "Message": "The field `email` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 11,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "email",
        "responseName": "email",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    },
    {
      "Message": "The field `deliveryInstructions` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 12,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "deliveryInstructions",
        "responseName": "deliveryInstructions",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    },
    {
      "Message": "The field `items` does not exist on the type `OrderConnection`.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "orders"
      },
      "Locations": [
        {
          "Line": 13,
          "Column": 33
        }
      ],
      "Extensions": {
        "type": "OrderConnection",
        "field": "items",
        "responseName": "items",
        "specifiedBy": "http://spec.graphql.org/June2018/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      },
      "Exception": null
    }
  ],
  "Extensions": null,
  "ContextData": {
    "HotChocolate.Execution.ValidationErrors": true
  },
  "HasNext": null
}
